mlflow/tracing/display/display_client.py

```diff
@@ -0,0 +1,78 @@
+import json
+from typing import List
+
+from mlflow.entities import Trace
+
+
+def _serialize_trace_list(traces: List[Trace]):
+    return json.dumps(
+        # we can't just call trace.to_json() because this
+        # will cause the trace to be serialized twice (once
+        # by to_json and once by json.dumps)
+        [json.loads(trace.to_json()) for trace in traces]
+    )
+
+
+class IPythonTraceDisplayClient:
+    _instance = None
+
+    @classmethod
+    def get_instance(cls):
+        if cls._instance is None:
+            cls._instance = IPythonTraceDisplayClient()
+        return cls._instance
+
+    def __init__(self):
+        self._prev_execution_count = -1
+        self.display_handle = None
+        self.traces_to_display = {}
+
+    def get_mimebundle(self, traces: List[Trace]):
+        if len(traces) == 1:
+            return traces[0]._repr_mimebundle_()
+        else:
+            return {
+                "application/databricks.mlflow.trace": _serialize_trace_list(traces),
+                "text/plain": traces.__repr__(),
+            }
+
+    def display_traces(self, traces: List[Trace]):
+        # this should do nothing if not in an IPython environment
+        try:
+            from IPython import get_ipython
+            from IPython.display import display
+
+            if len(traces) == 0 or get_ipython() is None:
+                return
+
+            traces_dict = {trace.trace_info.request_id: trace for trace in traces}
+
+            # if the current ipython exec count has changed, then
+            # we're in a different cell (or rerendering the current
+            # cell), so we should create a new display handle.
+            current_exec_count = get_ipython().execution_count
+            if self._prev_execution_count != current_exec_count:
+                self._prev_execution_count = current_exec_count
+                self.traces_to_display = traces_dict
+                self.display_handle = None
+            else:
+                self.traces_to_display.update(traces_dict)
+
+            deduped_trace_list = list(self.traces_to_display.values())
+            if self.display_handle:
+                self.display_handle.update(
+                    self.get_mimebundle(deduped_trace_list),
+                    raw=True,
+                )
+            else:
+                self.display_handle = display(
+                    self.get_mimebundle(deduped_trace_list),
+                    display_id=True,
+                    raw=True,
+                )
+        except Exception:
+            # swallow exceptions. this function is called as
+            # a side-effect in a few other functions (e.g. log_trace,
+            # get_traces, search_traces), and we don't want to block
+            # the core functionality if the display fails.
+            pass
```

Comment:
nit: Can we show warning or info here? I remember it was a bit challenging to debug the "trace not displayed" issue during bug-bash:)

Reply:
Let's use _logger.debug() to render the exception